arex-instrumentation/database/arex-database-mybatis3/src/main/java/io/arex/inst/database/mybatis3/ExecutorWrapper.java

```diff
@@ -82,6 +99,51 @@ private <U> U call(MappedStatement ms, Object o, BoundSql boundSql, ThrowingSupp
         return result;
     }
 
+    private void restoreKeyHolder(MappedStatement ms, DatabaseExtractor executor, Object o) {
+        try {
+            Reflector reflector = new Reflector(o.getClass());
+            String[] keyHolderList = StringUtils.split(executor.getKeyHolder(), KEYHOLDER_SEPARATOR);
+            String[] keyProperties = ms.getKeyProperties();
+
+            if (keyHolderList == null || keyProperties == null) {
+                return;
+            }
+
+            if (keyProperties.length != keyHolderList.length) {
+                return;
+            }
+            for (int i = 0; i < keyHolderList.length; i++) {
+                String[] valueType = StringUtils.split(keyHolderList[i], KEYHOLDER_TYPE_SEPARATOR);
+                Object keyHolderValue = SerializeUtils.deserialize(valueType[0], valueType[1]);
+                reflector.getSetInvoker(keyProperties[i]).invoke(o, new Object[]{keyHolderValue});
+            }
+        } catch (Throwable ex) {
+            LogUtil.warn("restoreKeyHolder failed.", ex);
+        }
+    }
+
+    private void saveKeyHolder(MappedStatement ms, DatabaseExtractor executor, Object o) {
+        try {
+            StringBuilder builder = new StringBuilder();
+            Reflector reflector = new Reflector(o.getClass());
+            for (String keyHolderName : ms.getKeyProperties()) {
+                Object keyHolderValue = reflector.getGetInvoker(keyHolderName).invoke(o, null);
```

Comment:
Make sure the return value of   `Object keyHolderValue = reflector.getGetInvoker(keyHolderName).invoke(o, null);` is non null before calling `keyHolderValue.getClass().getName()`